src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/UnstructuredAddress.java

```diff
@@ -40,12 +43,20 @@ public static UnstructuredAddress fromString(final String edifact) {
             System.arraycopy(splitByColon, 1, addressLines, 0, splitByColon.length - 1);
         }
 
-        final String postCode = splitByPlus[INDEX_POSTCODE];
+        final String postCode = arrayGetSafe(splitByPlus, INDEX_POSTCODE).orElse(null);
         final String qualifier = edifact.startsWith(KEY + PLUS_SEPARATOR + PLUS_SEPARATOR + FORMAT) ? FORMAT : "";
 
         return new UnstructuredAddress(qualifier, addressLines, postCode);
     }
 
+    public Optional<String[]> getAddressLines() {
+        return Optional.ofNullable(addressLines);
+    }
+
+    public Optional<String> getPostCode() {
```

Comment:
Minor thing, but can we rename this method (and the variable) to `getPostcode` and `postcode`? The "code" part shouldn't be capitalised

Reply:
No problem. Thanks!
18076f4059d052011eb4007e3c11e41ab0beff9f